### ISSUE TITLE
[BUG] Songs that use Sand, Gravel

### ISSUE THREAD

**Issue by TheAlienDrew** [HielkeMinecraft/OpenNoteBlockStudio#29]
When exporting a song, it works fine, however... with how Minecraft works, if you have a large enough song, some note blocks have sand and gravel, but are in the air... When playing the song, the sand or gravel (or any other gravity following block) will fall after the note block plays.

The fix? Include a solid block under any blocks that can fall.

**Comment by TheAlienDrew** [comment]
Also, here's a schematic to test... it has some sand that will fall as soon as a note block is played (in the zip).
[Coldplay - Clocks.zip](https://github.com/HielkeMinecraft/OpenNoteBlockStudio/files/3219609/Coldplay.-.Clocks.zip)

**Comment by HielkeMinecraft** [comment]
Are you sure there is always room for a block below the gravel/sand?

**Comment by TheAlienDrew** [comment]
Actually I'm not entirely sure, but I'm sure a check to see if a block is
underneath it could be done, to prevent replacing another noteblock.

**Comment by Creepercdn** [comment]
I have encountered similar problems.
Most of the time, there is a block of space under the sand and each time the note box is activated, it falls onto another note box.

**Comment by HielkeMinecraft** [comment]
Placing a block under the sand or gravel won't be a solution. When working with multiple layers, there are note blocks under the top ones. So placing a support block under the top ones would mute the lower note block's sound.

As you can see here: (bedrock = support)
![2019-07-28_20 17 14](https://user-images.githubusercontent.com/42968178/62011133-be631d00-b174-11e9-9689-a88ecc7247bf.png)

This issue used to be not a problem, because clay would make the same sound as gravel/sand. But alas, here we are.

Only way to fix this is to write it in the note block's block state. This does persist after being powered by redstone, but it will reset when the block under it is broken. Not bad.

2 problems with this approach.

* Minecraft
You can't store instrument type before Minecraft 1.13. So <1.13 you **must** set the block below the note block. 
We can fix this by just using clay when using Minecraft 1.12 and below.

* The .schematic format
This format hasn't been updated in years, so it won't save stuff like block states >1.13.
We can resolve this by ditching .schematic and switching over to the newer .schem format. This newer format is from worldedit itself and supports the latest versions.


Another way to fix this is by making the schematic file bigger and having more space between the note blocks. This is kind of a dirty fix and I don't think this is the best way to go.

**Comment by Creepercdn** [comment]
You can try to sign.

**Comment by TheAlienDrew** [comment]
Instead of using solid blocks then why not use something like a sign to
hold the sand in place? It would work and allow the blocks underneath to
play. you would just have to make sure the block state of the sign is set
to be on the side of the stone or whatever block is being used for the
frame of the note blocks.

**Comment by TheAlienDrew** [comment]
Or another idea is having slime blocks being used to keep the falling
blocks in place? (of course though I know the speed limit older versions of
the game)

**Comment by HielkeMinecraft** [comment]
Signs don't work. I placed a sign above a note blocks, and it still wasn't able to play.

And slime blocks don't work either, because falling blocks will fall even if they are attached to slime.

**Comment by TheAlienDrew** [comment]
Well... Looks like you're going to have to fix it the long way sometime
then...

>

**Comment by HielkeMinecraft** [comment]
The bad part about switching formats is that the schematics won't work with MCEdit anymore... 
But that's what you get when you don't update your program in years.

**Comment by AlsoScratch** [comment]
> The bad part about switching formats is that the schematics won't work with MCEdit anymore...
> But that's what you get when you don't update your program in years.

What's the problem with WorldEdit?

**Comment by HielkeMinecraft** [comment]
It only uses the .schematic format.

I've discovered that switching to .schem isn't an option anyway, because older versions of WorldEdit only support .schematic. Ugh.
This is getting real difficult.

**Comment by AlsoScratch** [comment]
What about if I am using the latest version of Minecraft and WorldEdit? (Just assuming)

**Comment by HielkeMinecraft** [comment]
That would work just fine, but then schematic export would only work for Minecraft 1.13 and up. And I think people aren't too thrilled with that.

**Comment by Creepercdn** [comment]
> That would work just fine, but then schematic export would only work for Minecraft 1.13 and up. And I think people aren't too thrilled with that.

Can you create a new feature to create a structure for Minecraft above 1.13?

**Comment by HielkeMinecraft** [comment]
Yeah that's possible. But it would make it kind of confusing for people. (.schematic <1.13 and .schem >1.13)  There must be a better way. It's just better to do everything the same way.

**Comment by untuned** [comment]
Why not increase the distance between layers, and then add a support block?

**Comment by TheAlienDrew** [comment]
The reason that would be some faulty thinking is, what if someone has all
minecraft instruments in a song??? Then the spacing apart from each layer
would be a bit too much... Though, horizonally, if we zig-zag'd the lines
so it would be like a zipper like fashion, adding a support block shouldn't
conflict.

>

**Comment by Quent12b0** [comment]
I've got the same issue, but I use this command block system. (French video)

https://www.youtube.com/watch?v=jLQWplYyKfY

**Comment by Quent12b0** [comment]
Other solution: place your percussion layer at the top on the studio (layer 1 or 2).

**Comment by Yanchen-wang** [comment]
instead of the compact note block layers make an option to make a flat plane where there are several different lines of noteblocks, or even multiple plains

**Comment by TheAlienDrew** [comment]
That would be an easy thing to implement too.

**Comment by Pepe20129** [comment]
Something to consider would be to use the .litematic format for schematics in 1.13+ as [Litematica](https://www.curseforge.com/minecraft/mc-mods/litematica) is a mod that has widely replaced Schematica in 1.13+, it supports block states & the MCEdit/Schematica .schematic format with support for the Worldedit/Sponge format coming soon.

**Comment by chenxi050402** [comment]
Using structure blocks won't solve the issue because the structure block updates all note block instruments when loading in.
